tensorrt_llm/_torch/pyexecutor/model_engine.py

```diff
@@ -1216,15 +1213,15 @@ def _prepare_tp_inputs(
             assert spec_dec_mode.support_overlap_scheduler(
             ), f"{self.spec_config.spec_dec_name} does not support overlap scheduler"
 
-        # will contain previous batch incices of generation requests
+        # will contain previous batch indices of generation requests
         previous_batch_indices = []
         previous_pos_indices = []
         for request in extend_requests:
             # the request has no previous tensor:
             # (1) next_draft_tokens_device is None, which means overlap scheduler is disabled; or
             # (2) a dummy request; or
             # (3) the first step in the generation server of disaggregated serving
-            if next_draft_tokens_device is None or request.is_dummy or request.py_batch_idx is None:
+            if not is_mtp or request.is_dummy or request.py_batch_idx is None:
```

Comment:
Should this take into account Eagle-3 as well?

Reply:
In that case, `is_mtp` is not a good name, as Eagle-3 and MTP are different things. Perhaps `is_spec_decoding`?